src/index.js

```diff
@@ -448,7 +448,8 @@ class Libp2p extends EventEmitter {
         config = { ...config, ...this._config.peerDiscovery[DiscoveryService.tag] }
       }
 
-      if (config.enabled) {
+      if (config.enabled &&
+        !this._discovery.filter((service) => service.tag === DiscoveryService.tag).length) { // not already added
```

Comment:
Should we just make `this._discovery` a Map? They require a tag, it might be easier for checking existence and getting specific ones should we need to do that.